### DISCUSSION TITLE
Supply both GNU libc and Musl Dockers?

### DISCUSSION THREAD

**Discussion by frno7** [frno7/mipsr5900el-gentoo-linux-gnu#4] [Ideas]
@AKuHAK and @immolo, maybe this repo could supply two Dockers, one with [GNU libc](https://en.wikipedia.org/wiki/Glibc) and another with [Musl](https://en.wikipedia.org/wiki/Musl)? Perhaps via two separate branches, `gentoo-gnu` and `gentoo-musl`, for instance. Renaming the repo once again, to simply `mipsr5900el-linux`?

**Comment by AKuHAK** [comment]
AFAIK you can hold only one package per repo. It is possible to differ them via different tags, but I think that it is better touse tags for tagging stable versions. So, my opinion: create different repofor musl package.

**Comment by immolo** [comment]
That's how I understand it too, I suggest we focus on glibc first though to make our lives easier.

**Comment by frno7** [comment]
A simpler alternative is to add a line `crossdev --stage4 --target mipsr5900el-unknown-linux-musl && \` just below

https://github.com/frno7/mipsr5900el-gentoo-linux-gnu/blob/9e01f44b0d5e04a496c8ac27e13494bf208eb63a/Dockerfile#L18

to have both GNU libc and Musl within the same Docker. Assuming Musl fits within the allotted space, this seems appealing. One could easily make two corresponding flavours of initramfs with them.

**Reply by immolo** [reply]
@frno7 I know this is possible on a server built to cross build multiple architectures however I haven't use docker before so I don't know that will work on such a system as I know chrooted builds have issues with it.

**Reply by frno7** [reply]
I just got to fiddle with Docker at Github. Normally I prefer [`schroot`](https://wiki.debian.org/Schroot). :-) The [crossdev](https://wiki.gentoo.org/wiki/Crossdev) package operates within Gentoo, so I don’t expect any particular problems with Docker. But separately I had Docker produce ppc64le and aarch64 tools, in addition to x86_64 which I don’t use myself. See https://github.com/frno7/iopmod/pull/9#issuecomment-1114194284 for additional details.

**Reply by immolo** [reply]
Well everyone tells me I need to switch to docker so maybe this will be excuse I need.

First though I think I should continue working with the Gentoo team about getting the official stage3 build done as there still seems to be an issue and then do the creation of the binhost for the packages the PS2 will struggle with as these are going to be the things I need do while I setup up my own environment, feel free to assign those tasks to me on the issues tracker as I'm already talking to Dilfridge on a regular bases for the first anyway.

**Reply by frno7** [reply]
Schroot works well with [QEMU/R5900](https://github.com/frno7/qemu). I’m going to amend my [app-emulation/qemu-mipsr5900el](https://github.com/frno7/gentoo.overlay/tree/master/app-emulation/qemu-mipsr5900el) package to have it install the appropriate [binfmt_misc](https://en.wikipedia.org/wiki/Binfmt_misc) handler too. That combination is _very_ convenient, and nice with a [Gentoo cross build environment](https://wiki.gentoo.org/wiki/Cross_build_environment), and testing initramfs, etc.

**Reply by immolo** [reply]
From my understanding doing that will mirror how the Gentoo MIPS team build the stage3 tarballs so having this ready could be a huge benefit.

**Comment by frno7** [comment]
I’m considering renaming this repo to something like `mipsr5900el-gentoo`. Not only for hosting both GNU libc and Musl, but also for relocating all Gentoo related articles from the [Linux wiki](https://github.com/frno7/linux/wiki) (and have it focus on kernel topics, as opposed to user applications, tooling, etc.). I mentioned this to @Arch91 some weeks ago.

**Reply by immolo** [reply]
I agree with you this would be a good idea as it will keep things cleaner.

**Reply by immolo** [reply]
@frno7 is it possible to move issues between repos or do they need to be recreated here instead?

**Reply by frno7** [reply]
Everything comes along to my understanding. Previous URLs redirect too.

**Reply by immolo** [reply]
Sorry I wasn't clear I meant the ones in the kernel repo which would be better placed in here once the rename happens.

**Reply by frno7** [reply]
https://docs.github.com/en/enterprise-cloud@latest/issues/tracking-your-work-with-issues/transferring-an-issue-to-another-repository

**Reply by immolo** [reply]
Excellent! As I don't have the necessary permissions to do this could you do it for me please?

**Reply by frno7** [reply]
Oh, sure.

**Reply by frno7** [reply]
Done, see #8!

**Comment by frno7** [comment]
Commit 9e1180913b36134d49a299da3646b21b7e663cbc adds Musl to the Docker, with the previously mentioned R5900 patch. Busybox now comes in both flavours:

https://github.com/frno7/mipsr5900el-gentoo-linux-gnu/blob/9e1180913b36134d49a299da3646b21b7e663cbc/Dockerfile#L39-L40

IOP tools and modules now install simply too:

https://github.com/frno7/mipsr5900el-gentoo-linux-gnu/blob/ec79a336b384b17204f9929663f2d4207e87ed87/Dockerfile#L45-L46
